snapshots/devmapper/metadata.go

```diff
@@ -108,8 +113,37 @@ func (m *PoolMetadata) AddDevice(ctx context.Context, info *DeviceInfo) error {
 
 		info.DeviceID = deviceID
 
-		return putObject(devicesBucket, info.Name, info, false)
+		return putObject(devicesBucket, info.Name, info, true)
```

Comment:
Hi,
correct me if I'm wrong here. If the existing device is "Faulty", its record in DB will be overwritten here, so that the dev id in this faulty record will be not tracked anymore - leaked?

Reply:
There are 2 states - device and devmapper device ID. Here we mutate *device* state (which is `getNextDeviceID` + `putObject`). Old device ID will remain in faulty state (have a look on `MarkFaulty` - https://github.com/containerd/containerd/pull/3470/files#diff-4132e94f99d36d21b09c446fb20687faR146

Here is an example:
1. Before
```
DevName=X ID=9 State=Faulty
ID=9 State=Faulty
```
2. After
```
DevName=X ID=10 State=OK
ID=9 State=Faulty
ID=10 State=OK
```

Reply:
Oh, I see, thanks.